flink-staging/flink-streaming/flink-streaming-core/src/test/java/org/apache/flink/streaming/api/functions/SocketClientSinkTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.functions;
+
+import java.io.IOException;
+import java.net.Socket;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.functions.sink.SocketClientSink;
+import org.apache.flink.streaming.util.serialization.SerializationSchema;
+
+import org.junit.Test;
+
+import static java.lang.Thread.sleep;
+import static org.junit.Assert.*;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+
+/**
+ * Tests for the {@link org.apache.flink.streaming.api.functions.sink.SocketClientSink}.
+ */
+public class SocketClientSinkTest{
+
+	private final String host = "127.0.0.1";
+	private int port = 9999;
+	private String access;
+	public SocketServer.ServerThread th = null;
+
+	class SocketServer extends Thread {
+
+		private ServerSocket server = null;
+		private Socket sk = null;
+		private BufferedReader rdr = null;
+		private PrintWriter wtr = null;
+
+		private SocketServer(int port) {
+			while (port > 0) {
+				try {
+					this.server = new ServerSocket(port);
+					break;
+				} catch (Exception e) {
+					--port;
+					if (port > 0) {
+						continue;
+					}
+					else{
+						e.printStackTrace();
+					}
+				}
+			}
+		}
+
+		public void run() {
+			System.out.println("Listenning...");
+			try {
+				sk = server.accept();
```

Comment:
This is prepare to subsequent complex test.
And I would like to remove this thread if you consider it`s not necessary.